python/tvm/topi/cuda/nms.py

```diff
@@ -575,6 +593,10 @@ def nms_inner_loop(ib, j):
                         nms_inner_loop(ib, j)
 
             num_valid_boxes[i] = num_valid_boxes_local[0]
+            # TODO(masahi): Want to do this instead of above, but the following is eliminated
+            # during codegen
+            # with ib.if_scope(tx == 0):
+            #     num_valid_boxes[i] = num_valid_boxes_local[0]
```

Comment:
Here, I want to generate the code like this:

```
if (threadIdx.x == 0) {
     num_valid_boxes[i] = num_valid_boxes_local[0]
}
```
but it seems `if_scope(tx == 0)` is eliminated during codegen. I guess this condition is considered a dead code by the simplifier? How can I generate code like this? @tqchen @Laurawly

Reply:
Same here from my understanding.